### PR TITLE
feat(vitest): export all reporters in `vitest/reporters`

### PR DESCRIPTION
**Pull request by Dunqing**
### Description

close: #3965

### Please don't delete this checklist! Before submitting the PR, please make sure you do the following:
- [x] It's really useful if your PR references an issue where it is discussed ahead of time. If the feature is substantial or introduces breaking changes without a discussion, PR might be closed.
- [x] Ideally, include a test that fails without this PR but passes with it.
- [x] Please, don't make changes to `pnpm-lock.yaml` unless you introduce a new test example.

### Tests
- [x] Run the tests with `pnpm test:ci`.

### Documentation
- [x] If you introduce new functionality, document it. You can run documentation with `pnpm run docs` command.

### Changesets
- [x] Changes in changelog are generated from PR name. Please, make sure that it explains your changes in an understandable manner. Please, prefix changeset messages with `feat:`, `fix:`, `perf:`, `docs:`, or `chore:`.

### CODE REVIEW

**Comment by sheremet-va** [comment]
You checked a checkbox that documentation is added, but I don't see any new documentation.

**Comment by Dunqing** [comment]
> You checked a checkbox that documentation is added, but I don't see any new documentation.

Sorry for that, where should I add documentation?

**Comment by sheremet-va** [comment]
> > You checked a checkbox that documentation is added, but I don't see any new documentation.
> 
> Sorry for that, where should I add documentation?

To the advanced section probably. Something like `Extending reporters`

File: docs/advanced/reporters.md [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+    // print something
+  }
+}
+```
+
+Then you can use your custom reporter in the `vitest.config.ts` file:
+
+```ts
+import { defineConfig } from 'vitest'
+import CustomReporter from './custom-reporter'
```
Comment (sheremet-va) [inline docs/advanced/reporters.md:50]:
```suggestion
import CustomReporter from './custom-reporter.js'
```

We are promoting valid ESM 😄

File: docs/advanced/reporters.md [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+
+export default defineConfig({
+  test: {
+    reporter: CustomReporter,
```
Comment (sheremet-va) [inline docs/advanced/reporters.md:54]:
```suggestion
    reporters: [new CustomReporter()],
```

`reporter` field is only supported for CLI

**Review by sheremet-va — CHANGES_REQUESTED** [review]

**Comment by Dunqing** [comment]
@sheremet-va You responded so fast, I haven't done a double-check yet 🤣

File: docs/advanced/reporters.md (hunk 1 above)
Comment (Dunqing) [reply docs/advanced/reporters.md:50]: But `./custom-reporter` is a ts file

Comment (sheremet-va) [reply docs/advanced/reporters.md:50]: It doesn't matter

Comment (Dunqing) [reply docs/advanced/reporters.md:50]: Ok, got it

**Pull request merged**